pkg/skaffold/kubernetes/port_forward.go

```diff
@@ -175,42 +181,101 @@ func (p *PortForwarder) portForwardPod(ctx context.Context, pod *v1.Pod) error {
 
 	for _, c := range pod.Spec.Containers {
 		for _, port := range c.Ports {
-			// If the port is already port-forwarded by another container,
-			// continue without port-forwarding
-			currentApp, ok := p.forwardedPorts[port.ContainerPort]
-			if ok && currentApp != c.Name {
-				color.LightYellow.Fprintf(p.output, "Port %d for %s is already in use by container %s\n", port.ContainerPort, c.Name, currentApp)
+			// get current entry for this container
+			entry, err := p.getCurrentEntry(pod, c, port, resourceVersion)
+			if err != nil {
+				color.Red.Fprintf(p.output, "Unable to get port for %s, skipping port-forward: %v", c.Name, err)
 				continue
 			}
-
-			entry := &portForwardEntry{
-				resourceVersion: resourceVersion,
-				podName:         pod.Name,
-				namespace:       pod.Namespace,
-				containerName:   c.Name,
-				port:            port.ContainerPort,
+			if entry.port != entry.localPort {
+				color.Yellow.Fprintf(p.output, "Forwarding container %s to local port %d.\n", c.Name, entry.localPort)
 			}
-
-			if prevEntry, ok := p.forwardedPods[entry.key()]; ok {
-				// Check if this is a new generation of pod
-				if entry.resourceVersion > prevEntry.resourceVersion {
-					p.Terminate(prevEntry)
-				}
+			if err := p.forward(ctx, entry); err != nil {
+				return errors.Wrap(err, "failed to forward port")
 			}
+		}
+	}
+	return nil
+}
 
-			color.Default.Fprintln(p.output, fmt.Sprintf("Port Forwarding %s %d -> %d", entry.podName, entry.port, entry.port))
-			p.forwardedPods[entry.key()] = entry
-			p.forwardedPorts[entry.port] = entry.containerName
+func (p *PortForwarder) getCurrentEntry(pod *v1.Pod, c v1.Container, port v1.ContainerPort, resourceVersion int) (*portForwardEntry, error) {
+	// determine if we have seen this before
+	entry := &portForwardEntry{
+		resourceVersion: resourceVersion,
+		podName:         pod.Name,
+		namespace:       pod.Namespace,
+		containerName:   c.Name,
+		port:            port.ContainerPort,
+	}
+	// If we have, return the current entry
+	oldEntry, ok := p.forwardedPods[entry.key()]
+	if ok {
+		entry.localPort = oldEntry.localPort
+		return entry, nil
+	}
+	// If another container isn't using this port...
+	if _, exists := p.forwardedPorts[port.ContainerPort]; !exists {
+		// ...Then make sure the port is availble
```

Comment:
linter is complaining s/availble/available/

Reply:
thanks!